### ISSUE TITLE
CBSE Maths grade 8 AI/ML activities

### ISSUE THREAD

**Issue by ndubbaka** [Pragament/coding_simulator_projects#6]
**For Class 8 students**, we can introduce more sophisticated AI/ML concepts that align with their growing mathematical maturity. Here are **AI/ML activities tailored to Class 8 topics**:

---

## **🧠 AI Concepts for Class 8 Level:**
- **Supervised vs Unsupervised Learning**
- **Training/Validation/Test splits**
- **Feature engineering**
- **Model evaluation metrics**
- **Simple neural networks**
- **Bias-Variance tradeoff**

---

### **Activity: "Speed-Distance-Time AI Model"**
1. **Create Dataset:**
   - Speed | Time → Distance (Direct: D = S×T)
   - Workers | Days → Work (Inverse: W×D = constant)
2. **"Train Different Models":**
   - **Model A:** Assumes direct proportion
   - **Model B:** Assumes inverse proportion
   - Test which fits better for given dataset
3. **Feature Engineering:** 
   - Create new feature: 1/Time for inverse relationship
   - Plot and compare R² values
4. **Real Application:** How GPS apps estimate arrival time

### **Extension:** Build simple regression in Google Sheets

---

### **Activity: "Algebraic Expression Classifier"**
1. **Create Expression Cards:**
   - x² + 5x + 6 → (x+2)(x+3)
   - x² - 9 → (x-3)(x+3)
   - 4x² - 25 → (2x-5)(2x+5)
2. **"Feature Extraction":**
   - Number of terms: 3 → try quadratic factorization
   - Two terms with squares → difference of squares
   - Perfect square pattern → (a±b)²
3. **Build Decision Tree:** Flowchart for factorization method
4. **Accuracy Test:** 20 expressions, track success rate

### **Tech Demo:** Show Symbolab/Wolfram Alpha factorization

---

### **Activity: "Shape Recognition using Euler's Formula"**
1. **Create Shape Database:**
   - Cube: V=8, E=12, F=6
   - Pyramid: V=5, E=8, F=5
   - Prism: V=6, E=9, F=5
2. **"Graph Features":** Represent shapes as connection graphs
3. **Train Classifier:** Given V, E, F → predict shape name
4. **Euler's Formula as Rule:** V - E + F = 2 (for polyhedrons)
5. **"Anomaly Detection":** Which shape doesn't follow formula?

### **AI Connection:** Social networks use similar graph analysis

---

### **Activity: "Optimal Graph Type Recommender"**
1. **Dataset Examples:**
   - Favorite subjects of 50 students
   - Test scores distribution
   - Monthly rainfall data
2. **"AI Rules Engine":**
   - If comparing parts of whole → Pie chart
   - If showing distribution → Histogram
   - If showing trends → Line graph
3. **Train with Examples:** Show data → student recommends chart type
4. **Accuracy Scoring:** Compare with correct choices

### **Real Tool:** Show how Google Sheets suggests charts

---

### **Activity: "Investment Growth AI"**
1. **Dataset Creation:**
   - Principal, Rate, Time, Final Amount
   - Use A = P(1 + r/n)^(nt)
2. **"Model Comparison":**
   - Linear model (wrong for compound interest)
   - Exponential model (correct)
3. **Visual Difference:** Plot both on graph
4. **Predict Future:** Given 5 years data, predict year 6, 7, 8
5. **Error Analysis:** Linear vs exponential error over time

### **Real World:** Stock prediction models (simplified)

---

### **Activity: "Number Type Detective"**
1. **Number Database:**
   - Rational: ⅓, 0.75, 2, -5, 0.333...
   - Irrational: √2, π, e, √3
2. **"Feature Engineering":**
   - Decimal expansion pattern
   - Square root or not
   - Can be expressed as fraction?
3. **Create Decision Rules:**
   ```
   If decimal terminates or repeats → Rational
   Else if square root of non-perfect square → Irrational
   Else → Need more analysis
   ```
4. **Confidence Scores:** 95% sure √4 is rational, 99% sure π is irrational

### **Extension:** Show how AI classifies emails as spam/not spam

---

### **Activity: "Visual Equation Solver"**
1. **Represent as Matrices:**
   ```
   2x + 3y = 8      becomes     [2 3] [x] = [8]
   4x - y = 2                   [4 -1][y]   [2]
   ```
2. **Graphical Solution:** Plot both lines, find intersection
3. **"AI Approaches":**
   - Graphical method (visual)
   - Substitution method (step-by-step)
   - Matrix method (efficient for computers)
4. **Compare Efficiency:** Time each method for 10 problems
5. **Special Cases:** Parallel lines (no solution), same line (infinite)

### **Tech Demo:** Show numpy solving equations in Python (concept only)

---

### **Activity: "Most Efficient Box Designer"**
1. **Problem:** Given 100cm² cardboard, design box with maximum volume
2. **"Brute Force AI":** Try all combinations (length, width, height)
3. **"Smart AI":** Use formula relationships
4. **Create Optimization Table:**
   - Length | Width | Height | Volume
   - Vary dimensions, calculate volume
5. **Find Pattern:** Square base gives better volume?
6. **Compare with Calculus solution:** (advanced preview)

### **Real World:** Package optimization in logistics

---

### **Activity: "Dice Roll Distribution Learner"**
1. **Collect Data:** Roll 2 dice 100 times, record sums (2-12)
2. **Theoretical Distribution:** Calculate expected probabilities
3. **Experimental Distribution:** Actual results from rolls
4. **"AI Learning Curve":**
   - After 10 rolls: uneven distribution
   - After 50 rolls: getting closer
   - After 100 rolls: close to theoretical
5. **Law of Large Numbers:** More data → better predictions

### **Extension:** Simulate in spreadsheet with RANDBETWEEN()

---

### **Activity: "Social Media Shares Predictor"**
1. **Hypothesis:** Shares follow power law (few posts get most shares)
2. **Create Model:** Shares = k × (Followers)^p
3. **Log Transformation:** Plot log(Shares) vs log(Followers) → should be linear
4. **Estimate Exponent p:** From graph slope
5. **Test Prediction:** For account with X followers, predict shares

### **Real Data:** YouTube views vs subscribers (simplified)

---

## **🛠️ Digital Tools for Class 8:**

1. **Google Colab** (Python notebooks with pre-written code)
2. **Orange Data Mining** (drag-drop visual programming)
3. **TensorFlow Playground** (neural network visualization)
4. **Google Sheets with ML features** (Explore function)
5. **AI for Oceans** (Code.org - ethical AI concepts)

---

### **Project: "Math Test Performance Predictor"**
1. **Data Collection:** Previous test scores, study hours, homework completion
2. **Feature Engineering:** 
   - Average score
   - Improvement trend
   - Study consistency
3. **Build Model:** Predict next test score range
4. **Validate:** Compare predictions with actual
5. **Present:** Dashboard with insights

### **Project: "Optimal Study Schedule AI"**
1. **Problem:** Limited time before exam
2. **Data:** Chapter difficulty, your strength, marks weightage
3. **Optimization:** Which chapters to focus on for maximum marks?
4. **Create Algorithm:** Prioritization system
5. **Test:** On previous exams (what would have been optimal?)

---

### **1. "AI Bias in Grading Systems"**
- If AI trained on boys' math papers only
- Would it be fair for grading girls?
- Importance of representative data

### **2. "AI vs Human Problem Solving"**
- Speed: AI faster for calculations
- Understanding: Humans better at novel problems
- Collaboration: Best of both worlds

### **3. "Design an AI Math Competition"**
- What tasks?
- How to score?
- Human vs AI categories

---

### **Neural Network Hands-On:**
```
Input Layer: [Number1, Operation, Number2]
Hidden Layer 1: [Detect if fraction, Detect if decimal, Detect if negative]
Hidden Layer 2: [Apply appropriate rules]
Output Layer: [Answer]
```

### **Training with Backpropagation Analogy:**
1. Make prediction
2. Check answer
3. Calculate error
4. Adjust "neuron connections" (rules)
5. Try again

---

## **✅ Assessment Ideas:**

1. **"Build a Classifier"**: For types of quadrilaterals with 85%+ accuracy
2. **"Dataset Creation"**: Collect and prepare data for an AI math tutor
3. **"Model Evaluation"**: Compare two approaches for same problem
4. **"Ethical Analysis"**: Case study on biased educational AI

---

## **🚀 Curriculum Integration Plan:**

**Month 1-2:** Activities 1-3 (Direct proportion, Factorization, Graphs)  
**Month 3-4:** Activities 4-6 (Data visualization, Compound interest, Classification)  
**Month 5-6:** Project work + Activities 7-8 (Equations, Optimization)  
**Month 7-8:** Advanced concepts + Ethics discussions

---

## **🌟 Quick Classroom Activities (60 mins):**

**Activity A:** "Will it Factor?" - Quick classification game  
**Activity B:** "Graph the AI" - Plot learning curves  
**Activity C:** "Optimization Challenge" - Best dimensions competition  
**Activity D:** "Probability Simulator" - Law of large numbers demo

These activities make **abstract Class 8 math concepts concrete** while introducing **real AI/ML principles** that prepare students for future STEM learning!
